pkg/common/configs/configvalidator.go

```diff
@@ -320,13 +328,17 @@ func checkLimit(limit Limit, currIdx int, userWildCardIdx, groupWildCardIdx *int
 		// It means the wildcard for user should be the last item for limits object list which including the username,
 		// and we should only set one wildcard user for all limits
 		if name == "*" {
-			if *userWildCardIdx != -1 && currIdx > *userWildCardIdx {
+			if limitCheckHelper.userWildCardIdx != -1 && currIdx > limitCheckHelper.userWildCardIdx {
 				return fmt.Errorf("should not set more than one wildcard user")
 			}
-			*userWildCardIdx = currIdx
-		} else if *userWildCardIdx != -1 && currIdx > *userWildCardIdx {
+			limitCheckHelper.userWildCardIdx = currIdx
+		} else if limitCheckHelper.userWildCardIdx != -1 && currIdx > limitCheckHelper.userWildCardIdx {
 			return fmt.Errorf("should not set no wildcard user %s after wildcard user limit", name)
 		}
+		if limitCheckHelper.existedUserName[name] {
+			return fmt.Errorf("duplicated user name %s , already existed", name)
+		}
+		limitCheckHelper.existedUserName[name] = true
```

Comment:
Should we move this block to the beginning? as duplicate check is expected to happen first ideally

Reply:
Good suggestion, addressed in latest PR.